tsfresh/feature_selection/selection.py

```diff
@@ -124,7 +136,141 @@ def select_features(X, y, test_for_binary_target_binary_feature=defaults.TEST_FO
 
         y = pd.Series(y, index=X.index)
 
-    df_bh = check_fs_sig_bh(X, y, n_jobs, chunksize, fdr_level, hypotheses_independent,
-                            test_for_binary_target_real_feature)
+    relevance_table = get_relevance_table(
+        X, y, ml_task=ml_task, n_jobs=n_jobs, chunksize=chunksize,
+        test_for_binary_target_real_feature=test_for_binary_target_real_feature,
+        fdr_level=fdr_level, hypotheses_independent=hypotheses_independent,
+    )
+
+    relevant_features = relevance_table[relevance_table.relevant].Feature
+
+    return X.loc[:, relevant_features]
+
+
+def infer_ml_task(y):
+    """
+    Infer the machine learning task to select for.
+    The result will be either `'regression'` or `'classification'`.
+    If the target vector only consists of integer typed values, we assume the task is `'classification'`.
+    Else `'regression'`.
+
+    :param y: The target vector y.
+    :type y: pandas.Series
+    :return: 'classification' or 'regression'
+    :rtype: str
+    """
+    if y.dtype.kind in np.typecodes['AllInteger']:
+        ml_task = 'classification'
+    else:
+        ml_task = 'regression'
+
+    _logger.warning('Infered {} as machine learning task'.format(ml_task))
```

Comment:
I'm open to changing it to `info` level. The thing is, our inference of the machine learning task is very opinionated and might be suprising for some. Imagine passing in housing prices as integers not knowing you end up performing multiclass selection.
Still, I think the inference is good this way. We just have to make it very clear to the user.